OpenRA.Mods.Common/Widgets/ActorPreviewWidget.cs

```diff
@@ -58,15 +58,9 @@ public void SetPreview(ActorInfo actor, TypeDictionary td)
 
 			var r = preview.SelectMany(p => p.ScreenBounds(worldRenderer, WPos.Zero));
 
-			if (r.Any())
-			{
-				var b = r.First();
-				foreach (var rr in r.Skip(1))
-					b = Rectangle.Union(b, rr);
-
-				IdealPreviewSize = new int2(b.Width, b.Height);
-				PreviewOffset = -new int2(b.Left, b.Top) - IdealPreviewSize / 2;
-			}
+			var b = r.Union();
+			IdealPreviewSize = new int2(b.Width, b.Height);
```

Comment:
Assuming `!r.Any()`, wouldn't `b` then be `Rectangle.Empty` which would then mean setting this to `(0, 0)`?
Can we safely assume there is always at least one entry in `r`?

Reply:
Good spot here - this is a behaviour change if `!r.Any()`

Thankfully, just a fraction above:
```c#
PreviewOffset = int2.Zero;
IdealPreviewSize = int2.Zero;
```

so even if it is empty (which would be odd) we're already doing what it wanted anyway. I've streamlined the code a touch as a result.

Other changes guarded by a `.Any` look similarly safe to drop that check in.